lib/date_range.rb

```diff
@@ -0,0 +1,56 @@
+require 'date'
+
+module Hotel
+  class DateRange
+    attr_reader :check_in, :check_out
+    
+    def initialize(check_in, check_out)
+      @check_in = check_in
+      @check_out = check_out
+      
+      validate_date_range
+      validate_date_object
+    end
+    
+    
+    def validate_date_range
+      if @check_out < @check_in
+        raise ArgumentError, "End date cannot be before start date"
+      end
+    end
+    
+    
+    def validate_date_object
+      unless @check_in.class == Date && @check_out.class == Date
+        raise ArgumentError, "Please enter a valid date!"
+      end
+    end    
+    
+
+    def num_nights
+      num_nights = (@check_out - @check_in).to_i
+      return num_nights
```

Comment:
You don't need to assign to a variable before returning a value:

```suggestion
      return (@check_out - @check_in).to_i
```